public/app/features/manage-dashboards/state/actions.ts

```diff
@@ -203,13 +203,16 @@ export function moveDashboards(dashboardUids: string[], toFolder: FolderInfo) {
 async function moveDashboard(uid: string, toFolder: FolderInfo) {
   const fullDash: DashboardDTO = await getBackendSrv().get(`/api/dashboards/uid/${uid}`);
 
-  if ((!fullDash.meta.folderId && toFolder.id === 0) || fullDash.meta.folderId === toFolder.id) {
+  if (
+    (fullDash.meta.folderUid === undefined && fullDash.meta.folderId === null && toFolder.uid === '') ||
+    fullDash.meta.folderUid === toFolder.uid
+  ) {
```

Comment:
Just out of curiosity, which scenarios does `folderUid === undefined && folderId === null`? is it some sort of legacy thing?

Reply:
Good eyes @JoaoSilvaGrafana it suppose to be `folderUid`. About the scenario, It seems part of a legacy thing, I'm not entirely sure which scenario it catches as it always go to the second condition(`fullDash.meta.folderUid === toFolder.uid`). 

Probably there might be a case where `folderUid` is `undefined` or `null` and if this happen the same as the destination folder is `''` which is general then we assume that they are in the same general folder location hence sending `{ alreadyInFolder: true };`